python_modules/dagster/dagster/core/execution.py

```diff
@@ -590,22 +659,17 @@ def _create_resources(pipeline_def, context_def, environment, execution_context,
         yield resources_type(**resources)
 
 
-def get_resource_or_gen(pipeline_def, context_definition, resource_name, environment, run_id):
-    resource_def = context_definition.resources[resource_name]
-    # Need to do default values
-    resource_config = environment.context.resources.get(resource_name, {}).get('config')
-    return resource_def.resource_fn(
-        InitResourceContext(
-            pipeline_def=pipeline_def,
-            resource_def=resource_def,
-            context_config=environment.context.config,
-            resource_config=resource_config,
-            run_id=run_id,
-        )
-    )
-
+def _execute_pipeline_iterator(context_or_failure_event):
```

Comment:
i guess the alternative was to throw out of the context manager and then put the yield context_or_failure_event in an except block. that might be more idiomatic in python

Reply:
im pretty sure that path involves wrapping a few different places in try/except where as handling it here is done once. That was my main motivation for this approach.